File: CD/CD.java
import java.util.HashSet;
import java.io.InputStreamReader;
import java.io.BufferedReader;

public class CD {
  public static void main(String[] args) throws Exception {
    BufferedReader scan = new BufferedReader(new InputStreamReader(System.in));
    HashSet<String> jackscds = new HashSet<String>(1000000);
    String[] parts = scan.readLine().split(" ");
    int n = Integer.parseInt(parts[0]);
    int m = Integer.parseInt(parts[1]);
    //System.out.println(n);
    //System.out.println(m);

    while(true){
      for (int i = 0; i < n; i++) {
        jackscds.add(scan.readLine());
      }

      int sum = 0;
      for (int i = 0; i < m; i++) {
        String cd = scan.readLine();
        if (jackscds.contains(cd)) {
          sum++;
        }
      }
      System.out.println(sum);
      parts = scan.readLine().split(" ");
      n = Integer.parseInt(parts[0]);
      m = Integer.parseInt(parts[1]);
      if (n == 0 && m == 0) {
        break;
      }
      jackscds.clear();
    }
  }
}

File: CD/cd.py
import io
import sys

file = [int(x) for x in sys.stdin.read().split()]
n = int(file[0])
m = int(file[1])
sum = 0
i = 2
while True:
    jackcds = set()
    for cd in range(n):
        jackcds.add(file[i])
        i +=1
    for cd in range(m):
        if cd in jackcds:
            sum +=1
        i +=1
    sys.stdout.write(str(sum) + "\n")
    n = file[i]
    m = file[i+1]
    if n == 0 and m == 0:
        break
    jackcds.clear()
